### PR TITLE
Issue_2121 added file DSA Documentation

### PR DESCRIPTION
**Pull request by newbiens**
<hr>

## Description 📜

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. 
Fixes # <#2121> 

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x ] Doc#umentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [ ] C/#CPP
- [ ] Competitive #Programming
- [ ] Cyber #Security
- [x ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ ] Java#Script
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [x ] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [ x] I have performed a self-review of my own code or work.
- [x ] I have commented my code, particularly in hard-to-understand areas.
- [ x] My changes generates no new warnings.
- [x ] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Review by sneha-nitdgp — CHANGES_REQUESTED** [review]
Some typos are present, fix them.

**Comment by newbiens** [comment]
Fixed

**Review by harshsri28 — CHANGES_REQUESTED** [review]

File: DSA/AnalysisOfSearchingAlgorithms/Searching Algorithm.txt [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+				else
+					low=mid+1;
+
+			}
+			return -1;
+
+
+		}
+
+Comparision between Binary Search and Linear Search
```
[the remark this answers is not in the document]
Comment (harshsri28) [reply DSA/AnalysisOfSearchingAlgorithms/Searching Algorithm.txt]: Remove unnecessary space also

**Comment by newbiens** [comment]
I have made the required changes.

**Review by harshsri28 — APPROVED** [review]
LGTM

**Review by sneha-nitdgp — APPROVED** [review]
LGTM

**Comment by nimisha-yadav** [comment]
Hey @newbiens, your PR is good but the issue you've created looks repetitive, I'd try to add your content too to the file and will let you know about the same.

Meanwhile, feel free to comment on the new issues I've created, you may work on them without any hassle.

**Comment by newbiens** [comment]
only a certain topic might be similar but the whole issue and its contents are completely new as far as I've seen, Please merge this one.

**Review by nimisha-yadav — COMMENTED** [review]
Looks great. Just rename your folder as 2.2 Searching & Sorting Algorithms and your file as 2.2_Overview_and_Standard_Problem_On_Binary_Search.md

**Comment by newbiens** [comment]
Done

**Review by nimisha-yadav — APPROVED** [review]
Congrats on getting your PR merged :star:

**Comment by newbiens** [comment]
Thanks⭐

**Pull request merged**